common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -300,6 +312,25 @@ private boolean isNonceRedirect(@NonNull final String url) {
         return url.contains(AuthenticationConstants.Broker.SSO_NONCE_PARAMETER);
     }
 
+    private boolean isCrossCloudRedirect(@NonNull final String url) {
+        try {
+            final URL currentUrl = new URL(url);
+            final URL startUrl = new URL(mRequestUrl);
+            if (AzureActiveDirectory.isValidCloudHost(currentUrl) && AzureActiveDirectory.isValidCloudHost(startUrl)) {
+                final Authority startUrlAuthority = Authority.getAuthorityFromAuthorityUrl(mRequestUrl);
+                final Authority currentLoadedUrlAuthority = Authority.getAuthorityFromAuthorityUrl(url);
+                if (!startUrlAuthority.getAuthorityURL().getHost().equalsIgnoreCase(currentLoadedUrlAuthority.getAuthorityURL().getHost())) {
+                    Logger.info(TAG, "Detected a cross cloud redirect.");
+                    return true;
+                }
+            }
```

Comment:
is this logic same as iOS's?

Reply:
Yes, iOS checks for 2 things
1. Is it a known cloud
2. If yes, then it checks if the host part of the startUrl and inputUrl are the same. 
![image](https://github.com/user-attachments/assets/777dda7c-75f5-4cdb-a1a0-3c79f1288625)